### PR TITLE
Added Binary Search [C]

### PR DESCRIPTION
**Pull request by singhpratyush**
#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

File: binary_search/binary_search.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * @arr_size Size of array
+ * @search_element Element to be searched
+ */
+int binary_search(const int *arr, int arr_size, int search_element) {
+	int left = 0, right = arr_size - 1;
+	while(left <= right) {
+		int mid = (left + right) / 2;
+		if(arr[mid] == search_element)	// Element found
+			return mid;
+		if(arr[mid] < search_element)	// Look in right half
```
Comment (Monal5031) [inline binary_search/binary_search.c]: I guess here it should be else if()

Comment (Monal5031) [inline binary_search/binary_search.c]: I think here it should be else if

Comment (singhpratyush) [inline binary_search/binary_search.c]: Logically, we already have an else there.

**Comment by singhpratyush** [comment]
Is it okay now?

**Comment by aviaryan** [comment]
@singhpratyush No. Please re-read my feedback.

**Comment by singhpratyush** [comment]
Oh! I couldn't understand, it was so deep. Will make the changes.

**Comment by singhpratyush** [comment]
@aviaryan : I have made the changes. Please review.

File: binary_search/binary_search.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * @arr_size Size of array
+ * @search_element Element to be searched
+ */
+int binary_search(const int *arr, int arr_size, int search_element) {
+	int left = 0, right = arr_size - 1;
+
+	while (left <= right) {
+		int mid = (left + right) / 2;
+
+		if(arr[mid] == search_element)	// Element found
```
Comment (aviaryan) [inline binary_search/binary_search.c]: Missing space after if . Same in the following `if`.

Comment (singhpratyush) [inline binary_search/binary_search.c]: Apologies for the miss. Please check once again.

**Comment by singhpratyush** [comment]
Was about to ask you to merge via command line. This is unpleasant to eyes. :laughing:

**Comment by singhpratyush** [comment]
And the squash was not needed I guess. They were just two commits.

**Comment by aviaryan** [comment]
@singhpratyush I think we should define a merge policy i.e. how a PR should be merged. If you agree, please go ahead and do so. 

> And the squash was not needed I guess. They were just two commits.

If I would have merged through command line, I wouldn't have squashed.

**Pull request merged**